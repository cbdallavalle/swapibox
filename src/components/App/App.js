import React, { Component } from 'react';
import swapiRepository from '../../helper.js';
import Nav from '../Nav/Nav';
import Main from '../Main/Main';
import './App.css';

class App extends Component {
  constructor() {
    super();
    this.state = {
      peopleToRender: [],
      vehiclesToRender: [],
      planetsToRender: [],
      swapiRepo: {},
      targets: [],
    }
  }

  componentDidMount() {
    const swapiRepo = new swapiRepository();
    this.setState({ swapiRepo })
  }

  renderPeople = async() => {
    if(!this.state.peopleToRender.length) {
      await this.state.swapiRepo.cleanPeopleData("https://swapi.co/api/people/?page=1"); 
    }
    this.setState({ peopleToRender: this.state.swapiRepo.people });
  }

  renderVehicles = async() => {
    if(!this.state.vehiclesToRender.length) {
      await this.state.swapiRepo.cleanVehicleData("https://swapi.co/api/vehicles/?page=1")
    }
    this.setState({ vehiclesToRender: this.state.swapiRepo.vehicles });
  }

  renderPlanets = async() => {
    if(!this.state.planetsToRender.length) {
      await this.state.swapiRepo.cleanPlanetData("https://swapi.co/api/planets/?page=1");
    }
    this.setState({ planetsToRender: this.state.swapiRepo.planets });
  }

  addTarget = (target) => {
    if(!(this.state.targets.find(obj => obj.name === target.name))) {
      const targets = [...this.state.targets, target];
      this.setState({targets})
    }
  }

  render() {
    return (
      <div className="App">
        <Nav renderPeople={ this.renderPeople } 
             renderVehicles={ this.renderVehicles }
             renderPlanets={ this.renderPlanets }
        />
        <Main peopleToRender={ this.state.peopleToRender } 
              vehiclesToRender={ this.state.vehiclesToRender }
              planetsToRender={ this.state.planetsToRender }
              targetsToRender={ this.state.targets }
              addTarget={ this.addTarget }
        />
      </div>
    );
  }
}

export default App;
