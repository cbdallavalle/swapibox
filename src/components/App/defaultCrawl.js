/* eslint-disable */
const defaultCrawl = {
  "title": "No Hope",
  "date": "1977-05-25", 
  "crawl": "It is a period of civil war.\r\nRebel scum spaceships, striking\r\nfrom a hidden base, have won\r\ntheir first and only victory against\r\nthe great Galactic Empire.\r\n\r\nDuring the battle, Rebel\r\nscum spies managed to steal secret\r\nplans to the Empire's\r\nultimate weapon, the DEATH\r\nSTAR, an armored space\r\nstation with enough power\r\nto destroy an entire planet.\r\n\r\nPursued by the Empire's\r\nfinest agents, Princess\r\nLeia flees home aboard her\r\nstarship, custodian of the\r\nstolen plans that can enslave her\r\npeople and restore\r\order to the galaxy...." 
};

export default defaultCrawl;