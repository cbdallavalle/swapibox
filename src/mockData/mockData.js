const mockData = {
  peopleData: {
    name: "Darth Vader",
    info: {
      homeworld: "Tatooine",
      species: "Human",
      homePop: "200000"  
    }

  },
  planetData: {
    name: "Alderaan", 
    info: {
      terrain: "grasslands, mountains", 
      population: "2000000000", 
      climate: "temperate", 
      residents: [
        {resident: "Leia Organa"},
        {resident: "Bail Prestor Organa"},
        {resident: "Raymus Antilles"}
      ]  
    }
  },
  vehicleData: {
    name: "Sand Crawler",
    info: {
      model: "Digger Crawler",
      passengers: "30",
      vehicleClass: "wheeled" 
    }
  },
}

export default mockData;