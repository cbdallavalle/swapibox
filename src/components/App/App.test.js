/* eslint-disable */
import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import App from './App';
import mockData from '../../mockData/mockData'

describe('App state', () => {
  let wrapper;

  it('default state should be an empty object', () => {
    wrapper = shallow(<App />, {disableLifecycleMethods: true});
    expect(wrapper.state()).toEqual(mockData.defaultState);
  })
})

describe('App methods', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App />);
  })
  
  it('should update state with an instance of swapiRepo on componentDidMount', () => {
    expect(wrapper.state()).toEqual(mockData.updateState);
  })

  it('should set state with cards to render if renderCards is called', async() => {
    await wrapper.instance().renderCards('vehicles');
    expect(wrapper.state().cardsToRender.length).toEqual(10)
  })

  it('should add or delete a target to the targets array if the target has or has not been previously marked', () => {
    wrapper.instance().toggleTarget(mockData.peopleData);
    expect(wrapper.state().targets.length).toEqual(1);

    wrapper.instance().toggleTarget(mockData.peopleData);
    expect(wrapper.state().targets.length).toEqual(0);
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})