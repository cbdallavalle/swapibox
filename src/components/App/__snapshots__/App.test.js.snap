// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App methods should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Nav
          getFilmCrawl={[Function]}
          renderCards={[Function]}
/>,
        <Main
          cardsToRender={Array []}
          filmCrawl={
                    Object {
                              "crawl": "It is a period of civil war.
                    Rebel scum spaceships, striking
                    from a hidden base, have won
                    their first and only victory against
                    the great Galactic Empire.
                    
                    During the battle, Rebel
                    scum spies managed to steal secret
                    plans to the Empire's
                    ultimate weapon, the DEATH
                    STAR, an armored space
                    station with enough power
                    to destroy an entire planet.
                    
                    Pursued by the Empire's
                    finest agents, Princess
                    Leia flees home aboard her
                    starship, custodian of the
                    stolen plans that can enslave her
                    people and restore
order to the galaxy....",
                              "date": "1977-05-25",
                              "title": "No Hope",
                            }
          }
          targetsToRender={Array []}
          toggleTarget={[Function]}
/>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "getFilmCrawl": [Function],
          "renderCards": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "cardsToRender": Array [],
          "filmCrawl": Object {
            "crawl": "It is a period of civil war.
Rebel scum spaceships, striking
from a hidden base, have won
their first and only victory against
the great Galactic Empire.

During the battle, Rebel
scum spies managed to steal secret
plans to the Empire's
ultimate weapon, the DEATH
STAR, an armored space
station with enough power
to destroy an entire planet.

Pursued by the Empire's
finest agents, Princess
Leia flees home aboard her
starship, custodian of the
stolen plans that can enslave her
people and restore
order to the galaxy....",
            "date": "1977-05-25",
            "title": "No Hope",
          },
          "targetsToRender": Array [],
          "toggleTarget": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Nav
            getFilmCrawl={[Function]}
            renderCards={[Function]}
/>,
          <Main
            cardsToRender={Array []}
            filmCrawl={
                        Object {
                                    "crawl": "It is a period of civil war.
                        Rebel scum spaceships, striking
                        from a hidden base, have won
                        their first and only victory against
                        the great Galactic Empire.
                        
                        During the battle, Rebel
                        scum spies managed to steal secret
                        plans to the Empire's
                        ultimate weapon, the DEATH
                        STAR, an armored space
                        station with enough power
                        to destroy an entire planet.
                        
                        Pursued by the Empire's
                        finest agents, Princess
                        Leia flees home aboard her
                        starship, custodian of the
                        stolen plans that can enslave her
                        people and restore
order to the galaxy....",
                                    "date": "1977-05-25",
                                    "title": "No Hope",
                                  }
            }
            targetsToRender={Array []}
            toggleTarget={[Function]}
/>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "getFilmCrawl": [Function],
            "renderCards": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "cardsToRender": Array [],
            "filmCrawl": Object {
              "crawl": "It is a period of civil war.
Rebel scum spaceships, striking
from a hidden base, have won
their first and only victory against
the great Galactic Empire.

During the battle, Rebel
scum spies managed to steal secret
plans to the Empire's
ultimate weapon, the DEATH
STAR, an armored space
station with enough power
to destroy an entire planet.

Pursued by the Empire's
finest agents, Princess
Leia flees home aboard her
starship, custodian of the
stolen plans that can enslave her
people and restore
order to the galaxy....",
              "date": "1977-05-25",
              "title": "No Hope",
            },
            "targetsToRender": Array [],
            "toggleTarget": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
